common/azureml_appinsights_logger/README.md

```diff
@@ -0,0 +1,147 @@
+[![Build Status](https://dev.azure.com/cse-manufacturing/MLOpsManufacturing/_apis/build/status/observability-library/observability-lib-ci?branchName=refs%2Fpull%2F59%2Fmerge)](https://dev.azure.com/cse-manufacturing/MLOpsManufacturing/_build/latest?definitionId=44&branchName=refs%2Fpull%2F59%2Fmerge)
+[![Build Status](https://dev.azure.com/cse-manufacturing/MLOpsManufacturing/_apis/build/status/observability-library/observability-lib-cd?branchName=paige%2Fobservability_lib)](https://dev.azure.com/cse-manufacturing/MLOpsManufacturing/_build/latest?definitionId=45&branchName=paige%2Fobservability_lib)
+
+# Overview
+Logs text and metrics in Azure ML, App Insights, and console with a single call. Also supports tracing in App Insights. 
+
+> Disclaimer: This package is built by Microsoft engineers, but it is not an officially supported Microsoft package.
+
+# How to use
+
+## 1. Add dependencies
+
+The following dependencies are required to use this package.
+
+```yaml
+  - python-dotenv==0.15.*
+  - dataclasses==0.*
+  - opencensus==0.7.*
+  - opencensus-context==0.1.*
+  - opencensus-ext-azure==1.0.*
+  - opencensus-ext-logging==0.1.*
+  - opencensus-ext-httplib==0.7.*
+```
+ 
+## 2. Configure environment variables to control logger behavior
+
+```bash
+# AppInsights logger is enabled if this is a valid connection string
+APPLICATIONINSIGHTS_CONNECTION_STRING = ''
+# Console logger is enabled if set to 'true'
+LOG_TO_CONSOLE = 'false'
```

Comment:
Should this be default to true?

Reply:
Hi @mokarian,

as I went deeper into the [python logging mechanism](https://docs.python.org/3/howto/logging.html#handlers) we are not using it very clean right now (opportunity for a refactoring/PR :) ). 

Actually we should just add a console/stream handler for console logging. 

The AppInsights logger works as a handler. So we [add the handler](https://github.com/liupeirong/MLOpsManufacturing/blob/0bd6873209cd4ff67d14318116bc196ccef39a30/common/azureml_appinsights_logger/azureml_appinsights_logger/appinsights_logger.py#L51) to the python logging and then calling the logging methods like [here](https://github.com/liupeirong/MLOpsManufacturing/blob/0bd6873209cd4ff67d14318116bc196ccef39a30/common/azureml_appinsights_logger/azureml_appinsights_logger/appinsights_logger.py#L102).

Even if several handlers are active you only need to call the `logger.<criticality>(msg)` once to get it recorded to all handlers. What happened when I did this with the console logger was due to our current structure [here](https://github.com/liupeirong/MLOpsManufacturing/blob/0bd6873209cd4ff67d14318116bc196ccef39a30/common/azureml_appinsights_logger/azureml_appinsights_logger/observability.py#L90) that `logger.<criticality>(msg)` was called twice  (once in the console logger object and then again in the app insights logger object) which resulted in doubled log entries on the console and on AppInsights, because 2 handlers were active at each point in time.

Therefore I just modified the console logger into almost the same thing as the AML logger, which just does a `print` to STDOUT. The only logger class which is currently using python `logger.<criticality>(msg)` is the App Insights class and contains only the App Insights handler, so we don't get duplicated messages into AppInsights.

However in the current situation, if you then run the scripts on AML service and the `LOG_TO_CONSOLE` is true, you get doubled STDOUT entries in the AML studio logger view. That's why I set default to `false` here.

The idea was that we wanted to get the library to life first and then take care about these details. There is room for improvement as you can see. I would think that we need following:

- rehaul the [Observability Class](https://github.com/liupeirong/MLOpsManufacturing/blob/0bd6873209cd4ff67d14318116bc196ccef39a30/common/azureml_appinsights_logger/azureml_appinsights_logger/observability.py#L51) to add handlers to the python logging mechanism as needed
- Calls to `logger.<criticality>(msg)` only happen once in Observability Class, rest is handled by the handlers
- AML Logger only needs to provide the metrics logging to AML service, the SDTOUT logging for AML Studio Log view can be captured by the Console Handler (I think it's called stream handler)

CC @liupeirong